tests/src/test/scala/org/http4s/parser/Rfc3986ParserSpec.scala

```diff
@@ -36,143 +35,142 @@ class Rfc3986ParserSpec extends Http4sSpec {
+      Ipv6Address.fromString(ipv6.value) must beRight(ipv6)
     }
 
     "parse ipv6 address with zeros in the fourth and fifth spots" in {
       val ipv6 = Ipv6Address(21494, -32768, 1, 0, 0, 1, 21787, 31704) // 53f6:8000:1::1:551b:7bd8
-      Rfc3986.ipv6.parse(ipv6.value).map(_._2) must beRight(ipv6)
+      Ipv6Address.fromString(ipv6.value) must beRight(ipv6)
     }
 
     "parse ipv6 address with zeros in the first six spots" in {
       val ipv6 = Ipv6Address(0, 0, 0, 0, 0, 0, 21787, 31704) // ::551b:7bd8
-      Rfc3986.ipv6.parse(ipv6.value).map(_._2) must beRight(ipv6)
+      Ipv6Address.fromString(ipv6.value) must beRight(ipv6)
     }
 
     "parse ipv6 address with zeros in the fifth and sixth spots" in {
       val ipv6 = Ipv6Address(21494, -32768, 1, 1, 0, 0, 21787, 31704) // 53f6:8000:1:1::551b:7bd8
-      Rfc3986.ipv6.parse(ipv6.value).map(_._2) must beRight(ipv6)
+      Ipv6Address.fromString(ipv6.value) must beRight(ipv6)
     }
 
     "parse ipv6 address with zeros in the first seven spots" in {
       val ipv6 = Ipv6Address(0, 0, 0, 0, 0, 0, 0, 31704) // ::7bd8
-      Rfc3986.ipv6.parse(ipv6.value).map(_._2) must beRight(ipv6)
+      Ipv6Address.fromString(ipv6.value) must beRight(ipv6)
     }
 
     "parse ipv6 address with zeros in the sixth and seventh spots" in {
       val ipv6 = Ipv6Address(21494, -32768, 1, 1, 21787, 0, 0, 31704) // 53f6:8000:1:1:551b::7bd8
-      Rfc3986.ipv6.parse(ipv6.value).map(_._2) must beRight(ipv6)
+      Ipv6Address.fromString(ipv6.value) must beRight(ipv6)
     }
 
     "parse ipv6 address with zeros in the all eight spots" in {
       val ipv6 = Ipv6Address(0, 0, 0, 0, 0, 0, 0, 0) // ::
-      Rfc3986.ipv6.parse(ipv6.value).map(_._2) must beRight(ipv6)
+      Ipv6Address.fromString(ipv6.value) must beRight(ipv6)
     }
 
     "parse ipv6 address with zeros in the seventh and eighth spots" in {
       val ipv6 = Ipv6Address(21494, -32768, 1, 1, 21787, 31704, 0, 0) // 53f6:8000:1:1:551b:7bd8::
-      Rfc3986.ipv6.parse(ipv6.value).map(_._2) must beRight(ipv6)
+      Ipv6Address.fromString(ipv6.value) must beRight(ipv6)
     }
 
     "parse ipv6 address with three zeros in a row" in {
       val ipv6 = Ipv6Address(22322, -32768, 24194, 32767, 0, 0, 0, 1) // 5732:8000:5e82:7fff::1
-      Rfc3986.ipv6.parse(ipv6.value).map(_._2) must beRight(ipv6)
+      Ipv6Address.fromString(ipv6.value) must beRight(ipv6)
     }
 
     "not parse an invalid ipv6 address containing an invalid letter such as 'O'" in {
       val invalidIp = "1200:0000:AB00:1234:O000:2552:7777:1313"
-      Rfc3986.ipv6.parse(invalidIp).map(_._2) must beLeft
+      Ipv6Address.fromString(invalidIp) must beLeft
     }
 
     "only parse part of an invalid ipv6 address where a single section is shortened (must be 2 or more)" in {
       val invalidIp = "2001:db8::1:1:1:1:1"
-      Rfc3986.ipv6.parse(invalidIp).map(_._2.value) must beRight("2001:db8::1:1:1:1")
+      Ipv6Address.fromString(invalidIp).map(_.value) must beLeft
```

Comment:
Clarified with @lewisjkl on #4004: they were Rights because the tests are using `parse`, and don't consume all input.  They flipped from left to right when the `Parser.end`s were removed.

`Ipv6Address.fromString` uses `parseAll`, which is a "clientside end", which correctly flips these back to lefts.  Jeff was right, and now so is this.